### ISSUE TITLE
Same 7-8 songs shuffle in the playlist

### ISSUE THREAD

**Issue by NakedFool** [AzuraCast/AzuraCast#1802]
Hey people,

So as the title says the same 7-8 songs in the playlist of 121 song shuffle each other.

I have read your known-issues - 

- Setting playlists to "random" instead of "shuffled"  - 
  It's set up shuffled from the start.

- The same song is in multiple albums or folders
  I've created folder for the each artists and checked inside each of them and there is no that kind of issue.

-  Frequent playlist changes or station restarts
Since i've played it few times i did restarted stream a few times, but it's causing problem in the long term as this does not change :/

-  Mis-weighted playlists
It is scheduled and as i considered it as a big one with 120 songs i set it up to 25.

So basically i'm not sure where to find solution, from the start they were all in music files without folders so i thought that may be. Then i created folders for each and situation is the same.

EDIT: I just found it in the duplicate section that there are duplicate songs ( once i created folder they appear there ) now i'm deleting those.

Do you think after deleting all of them problem should stop?

**Comment by radiolize** [comment]
You may try to deleting duplicate songs and try again and post the situation if continues here I think. Also you may want to check about differences between "Shuffled" and "Random" before you set.

**Comment by NakedFool** [comment]
Shuffled goes once through the playlist without repeating songs while random picks out random songs?

I just did this 1 hour ago and for now songs were not repeated, will inform the situation further if repeat happens.

With shuffled basically it should iterate once through the playlist without repeating any song in that playlist?

**Comment by radiolize** [comment]
Yep. Basicly it does not repeat until each song is played on playlist I think.

**Comment by roverradio** [comment]
I have this same exact issue on 0.9.6. Have around 250 songs in default playlist, and another playlist with sweepers playing every 2 songs.

When the default playlist is on Shuffle, it plays around 10 songs...then just repeats those same 10 songs in that order indefinitely.

**Comment by griffon56** [comment]
Hi, I have the same issue. I updated Azuracast last week and it is repeating the same tracks  of a playlist which contains 122 tracks. I use shuffled mode. Today I switched to sequential mode and it looks like the first tracks of the playlist (accessible by clicking ´reorder’ on the playlist) are the tracks that are played over and over.

**Comment by SlvrEagle23** [comment]
@griffon56 @NakedFool @roverradio If this issue is widespread, it's a high priority that I definitely need to investigate further. My local tests aren't showing anything out of the ordinary, so I will likely need to look at one of your installations to get to the bottom of the problem.

If one or more of you could join us on [our Mattermost team server](https://chat.azuracast.com/), you can direct message me there and we can discuss the specifics.

**Comment by SlvrEagle23** [comment]
@griffon56 @NakedFool @roverradio I've updated the AutoDJ to be more forgiving in how it handles playlists with either empty artists, or playlists that are composed of largely the same artist. This should significantly improve the issue where some playlists just repeat a small subset of contents over and over.

**Comment by roverradio** [comment]
Looking at this note. Yes, this playlist is a specialty stream - so all 250 songs are same artist, same album. Fits the pattern if that was indeed what was causing the issue. Thanks!

**Comment by griffon56** [comment]
Problem solved by the latest update - many thanks !

**Comment by griffon56** [comment]
Hi, sorry to comment on this closed topic - I am seeing the same issue with some songs that are repeated much more often that others. The playlist is shuffled. I saw that many of the songs in this playlist have the same name (=29). Could this be the cause of the issue ?

**Comment by griffon56** [comment]
I guess you would need to open a new topic since this one is now down the list. I am using v0.9.6.2 and don't see the same issue @roverradio describes anymore. I was just experiencing odd song repetition.
